assets/js/modules/analytics/util/parseDimensionStringToDate.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * convertDateStringToDateObject utility.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Converts a string date into an Date object and accounts for differences in timezones.
```

Comment:
Might be a good idea to reword this a bit in line with the name.

Also, we should include a `@since` annotation for all new functions. This one should be `@private` as well since we don't expose it publicly.